### ISSUE TITLE
Unable to login

### ISSUE THREAD

**Issue by iJasonWade** [XcodesOrg/XcodesApp#439]
**Describe the bug**
Upon trying to login an error message is thrown.

**To Reproduce**
Attempt to login with 2FA in place. (I use Yubico hardware keys tied to my iCloud account)

**Expected behavior**
Login asks for security key to be used.

**Screenshots**
![login-fail](https://github.com/XcodesOrg/XcodesApp/assets/12824479/4856375a-e336-44bf-952c-fcb3e36ca792)

**Version**
 - OS: Sonoma 14.0
 - Xcodes: 1.10.0 (18)

**Comment by easydev991** [comment]
I am not a developer of this fantastic app, but I can make a brief analysis of the problem: the error happens because there is a field missing in server response (`securityCode`), and currently it is expected to be available in your scenario

**Comment by czottmann** [comment]
I ran into the same problem. My account is also secured w/ a hardware key.

macOS 14.3.1 (23D60)
Xcodes 2.0.2 (23)

**Comment by czottmann** [comment]
@MattKiazyk I don't want to be rude or pushy, but is there any chance this might be looked at? It's a showstopper for me when it comes to using Xcodes. :(

**Comment by alexandregressier** [comment]
I'm experiencing the same issue with Xcodes version 2.0.2 (23) on macOS 14.5. Having used Xcodes (both GUI and CLI) for over two years without any problems, this is the first time I've encountered this. I'm not sure that's related, but I linked a hardware security key to my Apple ID more than two months ago. However, I'm fairly certain I've downloaded Xcode 15.3 before with this authentication setup.

Feel free to ask me any questions you think might be related to this problem.

![Screenshot 2024-05-19 at 10 29 49](https://github.com/XcodesOrg/XcodesApp/assets/13663677/b35ffaa3-b94e-42aa-8e62-e9b6fdd79fce)
